### DISCUSSION TITLE
Update/delete score using python sdk

### DISCUSSION THREAD

**Discussion by msanand** [langfuse/langfuse#1486] [Support] [answered]
Hi team,

I noticed that when I use langfuse.score() to create a new score for a traceid, it appends a new score to the trace even if the score with the same name exists already. I'm looking to implement a behavior where only one score is maintained for a trace with a given score name, and that score gets updated on change instead of a new score getting added to the trace. Using the python SDK I could not find an option to update an existing score for a given trace with a given score name. I also couldn't delete all existing scores before creating a new one. Can you please guide me on how I can implement this with the python SDK? Thanks!

**Comment by simone-costamagna** [comment, marked as answer]
You can use the DELETE API found here: https://api.reference.langfuse.com/#delete-/api/public/scores/-scoreId-

When you create a new score, the function will return a scoreID. Make sure to save this ID. If you need to update the score later, you'll first need to delete the previous score using the DELETE API, and then create a new one.

**Reply by mjrusso** [reply]
I noticed in [the docs](https://langfuse.com/docs/sdk/python/decorators#adding-scores) that you can provide an `id` argument to `score`, which has the following comment:

```
    id="unique_id" # optional, can be used as an indempotency key to update the score subsequently
```

I'm assuming that if you use the same ID, it has the same effect without needing to manually delete?

**Reply by mjrusso** [reply]
> I noticed in [the docs](https://langfuse.com/docs/sdk/python/decorators#adding-scores) that you can provide an `id` argument to `score`, which has the following comment:
> 
> ```
>     id="unique_id" # optional, can be used as an indempotency key to update the score subsequently
> ```
> 
> I'm assuming that if you use the same ID, it has the same effect without needing to manually delete?

Just tested this; looks like it works!

**Reply by FrankyCTY** [reply]
Yes, same for the API endpoint https://api.reference.langfuse.com/#post-/api/public/scores:~:text=the%20current%20limit-,Create%20a%20score,-POST%20/api/public

Internally, LangFuse is doing `upsert` operation based on if the score id is provided or not. If provided, it will try to perform update, otherwise it will create a new score.

**Comment by jasonslyvia** [comment]
We implemented this feature using API, it turns out the TS/JS SDK doesn't provide this feature either.

The use case is when user submitting a feedback by clicking thumb down icon in our product, we immediately create a score. There is an addtional step to provide detailed comment, if user provided, we'd need to update the previously created score.

**Reply by marcklingen** [reply]
You can achieve this by including an own `id` when creating the score. Thereby you can upsert it at a later point (e.g. when getting the comment by the user) with the user feedback by using the same `id` in the upsert.

**Comment by ranzeet013** [comment]
i am having trouble with the evaluators as well need some help
